frame/referenda/src/benchmarking.rs

```diff
@@ -41,17 +42,20 @@ fn funded_account<T: Config<I>, I: 'static>(name: &'static str, index: u32) -> T
 	caller
 }
 
-fn create_referendum<T: Config<I>, I: 'static>() -> (T::AccountId, ReferendumIndex) {
-	let caller = funded_account::<T, I>("caller", 0);
-	whitelist_account!(caller);
-	assert_ok!(Referenda::<T, I>::submit(
-		RawOrigin::Signed(caller.clone()).into(),
-		Box::new(RawOrigin::Root.into()),
-		T::Hashing::hash_of(&0),
-		DispatchTime::After(0u32.into())
-	));
+fn create_referendum<T: Config<I>, I: 'static>() -> (T::Origin, ReferendumIndex) {
+	let origin: T::Origin = T::SubmitOrigin::successful_origin();
+	if let Ok(caller) = frame_system::ensure_signed(origin.clone()) {
```

Comment:
what to do if `Err`?

Reply:
Nothing can be done if `Err` afaik, certainly cannot give a balance to an account :)

Reply:
Shouldn't the benchmarks fail or something like that?

Reply:
the benchmarks will fail if getting the origin / funds are not as expected. That being said, we can let the actual pallet call return the error versus assuming in the benchmark an error will occur.